### ISSUE TITLE
Proposal: U of Michigan Cavium Marvell Thunderx2 Phase 2 Big Data ML Testing

### ISSUE THREAD

**Issue by smeyer12** [WorksOnArm/cluster#157]
If you are interested in filing a request for access to the Works on Arm test and 
CI infrastructure, please fill out the details below, or contact Ed Vielmetti at ed@packet.net with questions.

If you are just making a comment, ignore/delete those fields and file your issue.

### Name, email, company, job title

Seth Meyer
smeyer@umich.edu
U of Michigan
Research Computing Lead

Marc Patton
pattonmj@umich.edu
U of Michigan
Research Computing Administrator

Jeremy Hallum
jhallum@umich.edu
U of Michigan
Research Computing Manager

Jeff Sica
jsica@umich.edu
U of Michigan
Research Database Manager

Bob Killen
rkillen@umich.edu
U of Michigan
Research Cloud Manager

Brock Palen
brockp@umich.edu
U of Michigan
Research Director

Note that projects with two or more participants are preferred.

### Project Title and description

The Marvell Phase 2 proposal is an initiative to build the second phase of our data science/HPC partnership  for researchers at the University of Michigan.   This proposal seeks to utilize the Marvell Thunder X2 processor for a blend of data science and High Throughput Computing (HTC) use cases. 

In a nutshell, we propose a hardware installation to support a kubernetes scheduler, which would submit work to a number of compute/storage nodes that run HDFS on them for scratch data storage.  Applications that run on this system would stretch from spark and pyspark applications, to machine learning applications such as tensorflow, SparkML/MLlib, pytorch, Keras, Theano, Caffe,  and others, that can be run inside of kubernetes, and, ideally, managed by Kubeflow.   HTC single node compute intensive workloads, often ran on HPC systems, could also be run on this machine as well within their own containers. 

The goals of this infrastructure are as follows:
To demonstrate the viability of ARM in the data center as a platform for Kubernetes and Kubeflow for research. 
To provide a safe, secure, and performant environment for researchers to perform advanced data analytics and machine learning techniques on research data. 
To provide an environment for the execution HTC workloads using containers on Kubernetes.

to attempt to perform advanced machine learning analytics on larger and larger datasets.  Through discussions with our researchers, more and more of them are working with patient data and large scale patient population studies that are outstripping their abilities to use past techniques and software prior (such as SAS, SPSS, R codes without spark).  As data sets grow, researchers needs access to advanced tools and technology, but also in a shared, secure environment that allows for analysis of the varied restricted data types that University Researchers have available to them.  Below is a diagram showing the various tooling and components we look to utilize and how we plan to use them.

A Kubernetes/Kubeflow cluster has not been deployed for research use using the ARM architecture, as far as we are aware.   Similar deployments to what we propose have been done in the x86 environments, this would be the first ARM64 deployment.  There could be a risk, as with phase 1, that there is some time before the ecosystem has matured on ARM sufficient for production use.  We hope to mitigate this risk by piloting the base level on the Packet.Net infrastructure run by Ed Vielmetti to proof out the final proposed hardware in the next two or three weeks.  

HTC use cases as well as Hadoop Ozone are in scope.

### Which members of the community would benefit from your work?

the k8s, kubeflow, Big data, hadoop, ML, AI, and deep learning communities.

### Is the code that you’re going to run 100% open source? 

Yes

Provide the URLs where it is located, or a Git repository.

TBD

### What infrastructure (computing resources and network access) do you need? (see: https://www.packet.net/bare-metal/)?

Thunderx2 hardware
amd64 possibly for prototyping

Let us know if you need short-term (one time) support, or if this is a request for
continuous ongoing support. If possible, please identify foundations or other
support organizations that can help with long-running projects.

### Describe the continuous integration (CI) system in use or desired for this project.

Are you testing every check-in on arm64? (Would you like to?) If so, share a public URL of a CI dashboard.

### Describe the monitoring system in use or desired for this project.

Are you running an agent on any long-running servers to track system status? Are you willing to share that data with the project? Examples might include a Prometheus metric endpoint or a Grafana dashboard.

### Please state your contributions to the open source community and any other relevant initiatives

Brag a little bit about yourself, please!

**Comment by vielmetti** [comment]
Reopen - the logistics on this is that we are queued on data center work behind the Qualcomm rack buildout.

In the mean time good discussions with @smeyer12 which should result in a punch list of containers we'd like to see running on arm64 as a piece of this project.

**Comment by vielmetti** [comment]
Reopening. We're looking into ThunderX2 firmware updates to make this ready.
